src/Raven.Server/Documents/Subscriptions/SubscriptionStorage.cs

```diff
@@ -160,6 +161,30 @@ public async Task AcknowledgeBatchProcessed(long subscriptionId, string name, st
                 DocumentsToResend = docsToResend
             };
 
+            var state = _serverStore.Engine.CurrentState;
+            if (command.ChangeVector == nameof(Constants.Documents.SubscriptionChangeVectorSpecialStates.DoNotChange) &&
+                (state == RachisState.Leader || state == RachisState.Follower))
+            {
+                // there are no changes for this subscription but we still want to check if we are the node that is responsible for this task.
+                // we can do that locally if we have a functional cluster (in a leader or follower state).
+
+                using (_serverStore.ContextPool.AllocateOperationContext(out TransactionOperationContext context))
+                using (context.OpenReadTransaction())
+                {
+                    var rawDatabaseRecord = _serverStore.Cluster.ReadRawDatabaseRecord(context, command.DatabaseName);
+                    if (rawDatabaseRecord == null)
+                        throw new DatabaseDoesNotExistException($"Cannot set command value of type {nameof(AcknowledgeSubscriptionBatchCommand)} for database {command.DatabaseName}, because it does not exist");
+
+                    var subscription = _serverStore.Cluster.Read(context, command.GetItemId());
+                    if (subscription == null)
+                        throw new SubscriptionDoesNotExistException($"Subscription with name '{command.SubscriptionName}' does not exist");
+
+                    var subscriptionTask = new AcknowledgeSubscriptionBatchCommand.SubscriptionTask(subscription);
+                    command.AssertSubscriptionState(rawDatabaseRecord, subscriptionTask, command.SubscriptionName);
+                    return;
```

Comment:
sending the command also fire trigger databases, where we check last subscription change vector, query and task state

https://github.com/ravendb/ravendb/blob/a63cc59621543258fcffe580de02f6bef9bc4fc0/src/Raven.Server/Documents/Subscriptions/SubscriptionStorage.cs#L674

Reply:
All of those are triggered when the subscription **does** change (disable subscription, different change vector...)
Sending the `AcknowledgeSubscriptionBatchCommand` with `DoNotChange` doesn't change the subscription.